include/hgwosca.h

```diff
@@ -6,97 +6,88 @@ extern "C" {
 
 #include "stddef.h"
 
-double *const hgwosca(double(*obj)(const double *const, size_t),
-                      size_t wolf_count,
-                      size_t dim,
-                      size_t max_iterations,
-                      const double *const min_positions,
-                      const double *const max_positions);
+double * hgwosca(double(*obj)(const double *const, size_t),
+                 size_t wolf_count,
+                 size_t dim,
+                 size_t max_iterations,
+                 const double *min_positions,
+                 const double *max_positions);
 
 /**
    Initialise population of `wolf_count` wolves, each with `dim` dimensions, where
    each dimension is bound by `min_positions` and `max_positions`.
  */
-double *const init_population(size_t wolf_count, size_t dim,
-                              const double *const min_positions,
-                              const double *const max_positions);
+double *init_population(size_t wolf_count, size_t dim, const double * min_positions, const double * max_positions);
 
 /**
    Update the fitness of all wolves.
  */
-void update_fitness(size_t wolf_count, size_t dim,
+void update_fitness(size_t wolf_count,
+                    size_t dim,
                     double(*obj)(const double *const, size_t),
-                    double *const population, double *const fitness);
+                    double *population,
+                    double *fitness);
 
 /**
    Find the new wolf leaders.
  */
-void update_leaders(size_t wolf_count, double *const fitness,
-                    size_t *const alpha,
-                    size_t *const beta,
-                    size_t *const delta);
-
+void update_leaders(size_t wolf_count, double *fitness, size_t *alpha, size_t *beta, size_t *delta);
 
 /**
    Initialise the fitness for all wolves.
  */
-double *const init_fitness(size_t wolf_count, size_t dim,
-                           double(*obj)(const double *const, size_t),
-                           double *const population);
-
+double *init_fitness(size_t wolf_count, size_t dim, double(*obj)(const double *const, size_t), double *population);
```

Comment:
All those functions could use the type definition, then a single change in the objective function calling structure might not require changes in all signatures (very error prone). This is just a suggestion though, I need to look more into the project, whether importing `benchmark.h` headers makes even sense, or if the type definitions would need to be moved to a different file.